relayer/app/app.go

```diff
@@ -61,7 +68,10 @@ func Run(ctx context.Context, cfg Config) error {
 			cprov,
 			xprov,
 			CreateSubmissions,
-			sendProvider)
+			sendProvider,
+			state)
```

Comment:
rather provide the cursors for this worker in the constructor. Setter should only be used if the data is not available at constructing time.

Reply:
I think you need to `CursorsForWorker(destChainID)` function